### PR TITLE
Expose Secrets in Source/SinkContext

### PR DESCRIPTION
**Pull request by srkukarni**
### Motivation
Currently Sources/Sinks can pass secrets via the Source/SinkConfig but the actual source/sinks implementations cannot use these secrets because the Source/SinkContext doesn't expose the secrets.
This pr exposes that interface.
### Modifications

*Describe the modifications you've done.*

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by srkukarni** [comment]
run cpp tests
run integration tests
run java8 tests

**Comment by sijie** [comment]
retest this please

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run java8 tests

File: pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/SchedulerManager.java [hunk 1]
```diff
@@ -105,25 +105,12 @@ public SchedulerManager(WorkerConfig workerConfig, PulsarClient pulsarClient, Pu
                     .compressionType(CompressionType.LZ4)
                     .sendTimeout(0, TimeUnit.MILLISECONDS)
                     .createAsync().get(10, TimeUnit.SECONDS);
+            } catch (Exception e) {
```
Comment (jerrypeng) [inline pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/SchedulerManager.java]: why is this change here?

Comment (srkukarni) [reply pulsar-functions/worker/src/main/java/org/apache/pulsar/functions/worker/SchedulerManager.java]: Oops, looks like bad commit. Reverted

**Comment by srkukarni** [comment]
run integration tests

**Comment by srkukarni** [comment]
run java8 tests

**Comment by srkukarni** [comment]
run integration tests
run java8 tests

**Comment by srkukarni** [comment]
run integration tests

**Comment by srkukarni** [comment]
run integration tests

**Comment by merlimat** [comment]
Merged in 2.3.1 at 
253381c
